ludwig/backend/ray.py

```diff
@@ -683,24 +691,40 @@ def get_resources_per_worker(self) -> Tuple[int, int]:
         num_cpus = resources_per_worker.get("CPU", (1 if num_gpus == 0 else 0))
         return num_cpus, num_gpus
 
-    def batch_predict(self, dataset: RayDataset, *args, collect_logits: bool = False, **kwargs):
+    def batch_predict(
+        self, dataset: RayDataset, *args, collect_logits: bool = False, model_ref: ObjectRef = None, **kwargs
+    ):
         self._check_dataset(dataset)
 
         predictor_kwargs = self.predictor_kwargs
         output_columns = get_output_columns(self.model.output_features, include_logits=collect_logits)
+
+        num_cpus, num_gpus = self.get_resources_per_worker()
+
+        distributed_strategy = self.trainer_kwargs.get("strategy", get_default_strategy_name())
+        if distributed_strategy == "deepspeed" or (
```

Comment:
Rather than do this for DeepSpeed specifically, it might make sense to do this for any LLM model type (we should also do it for LLMs as encoders, but at the moment they don't do model parallelism for inference, so it wouldn't work anyway).

Reply:
done